ravedude/src/avrdude/mod.rs

```diff
@@ -17,6 +17,50 @@ pub struct Avrdude {
 }
 
 impl Avrdude {
+    pub fn require_min_ver((req_major, req_minor): (u8, u8)) -> anyhow::Result<()> {
+        let output = process::Command::new("avrdude")
+            .arg("-?")
+            .output()
+            .context("Failed to start and gather avrdude output.")?;
+        let stderr: &str =
+            std::str::from_utf8(&output.stderr).context("Avrdude's stderr contains non-utf8.")?;
+        let version: &str = stderr
+            .split("version")
+            .last()
+            .context("Unable to obtain avrdude's version string")?
+            .trim();
+        let (version, _) = version
+            .split_once('-')
+            .context("Unable to obtain avrdude's version string")?
+            .to_owned();
+        let (major, minor) = version.split_once('.').unwrap();
+        let major = major
+            .parse::<u8>()
+            .context("Unable to parse major version number.")?;
+        let minor = minor
+            .parse::<u8>()
+            .context("Unable to parse minor version number.")?;
+        if major < req_major {
```

Comment:
Actually, you can combine the two `if`s because tuples in Rust implement comparison operations.  See this demonstration: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=836cd2133053e4b2e547af94a8fe8bda

Reply:
Really? Oh wow. Rust keeps on giving! ;)